internal/reconciliations/external_authorizer/external_authorizer_test.go

```diff
@@ -0,0 +1,240 @@
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func createFakeClient(objects ...ctrlclient.Object) ctrlclient.Client {
+	err := operatorv1alpha2.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = corev1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = networkingv1alpha3.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = securityv1beta1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = networkingv1beta1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	return fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(objects...).Build()
+}
+
+var _ = Describe("Reconciliation", func() {
+	const (
+		expectedName    = "expected-name"
+		expectedPort    = 1234
+		expectedService = "a.svc.cluster.local"
+	)
+
+	istioCR := operatorv1alpha2.Istio{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "default",
+			ResourceVersion: "1",
+			Annotations:     map[string]string{},
+			UID:             "1234",
+		},
+		Spec: operatorv1alpha2.IstioSpec{
+			Config: operatorv1alpha2.Config{
+				Authorizers: []*operatorv1alpha2.Authorizer{
+					{
+						Name:    expectedName,
+						Service: expectedService,
+						Port:    expectedPort,
+					},
+				},
+			},
+		},
+	}
+
+	Context("Duplicated authorization name", func() {
+		It("should cause a warning state", func() {
+			//given
+			client := createFakeClient()
+			reconciler := external_authorizer.NewReconciler(client)
+
+			istioCR := operatorv1alpha2.Istio{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "default",
+					ResourceVersion: "1",
+					Annotations:     map[string]string{},
+					UID:             "1234",
+				},
+				Spec: operatorv1alpha2.IstioSpec{
+					Config: operatorv1alpha2.Config{
+						Authorizers: []*operatorv1alpha2.Authorizer{
+							{
+								Name:    expectedName,
+								Service: expectedService,
+								Port:    expectedPort,
+							},
+							{
+								Name:    expectedName,
+								Service: expectedService,
+								Port:    expectedPort,
+							},
+						},
+					},
+				},
+			}
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
+
+			//then
+			Expect(err).To(HaveOccurred())
+			Expect(err.Description()).To(Equal(fmt.Sprintf("Authorizer name needs to be unique: %s is duplicated", expectedName)))
+		})
+	})
+
+	Context("ServiceEntry creation", func() {
+
+		It("should succeed creating ServiceEntry", func() {
+			//given
+			client := createFakeClient()
+			reconciler := external_authorizer.NewReconciler(client)
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
```

Comment:
I remember, that it was discussed to use context.Background instead of TODO. Is this still valid?